File: 696-count-binary-substrings/count-binary-substrings.go
func countBinarySubstrings(s string) int {
    if len(s) == 0{
        return 0
    }
    ans , prev , current := 0,0,1
    for i:= 1 ; i< len(s);i++{
        if s[i-1] != s[i]{
            if prev < current {
                ans += prev
            } else {
                ans += current 
            }
            prev = current 
            current = 1
        } else {
            current++
        }
    }
    
if prev < current {
        ans += prev
    } else {
        ans += current
    }
    return ans

}

File: 696-count-binary-substrings/count-binary-substrings.java
class Solution {
    public int countBinarySubstrings(String s) {
        int count = 0 , prev = 0 , current = 1;
        for(int i=1;i<s.length();i++){
            if(s.charAt(i-1) != s.charAt(i)){
                count += Math.min(prev,current);
                prev = current;
                current = 1;

            }else{
                current++;
            }
        }
        return count + Math.min(prev,current);
    }
}